pkg/crypto/rings/client.go

```diff
@@ -69,6 +72,84 @@ func (rc *ringClient) GetRingForAddress(
 	return newRingFromPoints(points)
 }
 
+// VerifyRelayRequestSignature verifies the relay request signature against the
+// ring for the application address in the relay request.
+func (rc *ringClient) VerifyRelayRequestSignature(
+	ctx context.Context,
+	relayRequest *types.RelayRequest,
+) error {
+	rc.logger.Debug().
+		Fields(map[string]any{
+			"session_id":          relayRequest.Meta.SessionHeader.SessionId,
+			"application_address": relayRequest.Meta.SessionHeader.ApplicationAddress,
+			"service_id":          relayRequest.Meta.SessionHeader.Service.Id,
+		}).
+		Msg("verifying relay request signature")
+
+	// Extract the relay request's ring signature
+	if relayRequest.Meta == nil {
+		return ErrRingClientEmptyRelayRequestSignature.Wrapf(
+			"request payload: %s", relayRequest.Payload,
+		)
+	}
+
+	signature := relayRequest.Meta.Signature
+	if signature == nil {
+		return sdkerrors.Wrapf(
+			ErrRingClientInvalidRelayRequest,
+			"missing signature from relay request: %v", relayRequest,
+		)
```

Comment:
Let's rewrite these error constructors to be less verbose:

```suggestion
		return ErrRingClientInvalidRelayRequest.Wrapf(
			"missing signature from relay request: %v", relayRequest,
		)
```